### PR TITLE
[GAT-8] Add test for govcms_social_links module.

### PR DESCRIPTION
**Pull request by NigelCunningham**

### CODE REVIEW

File: build/tests/behat/features/social-links.feature [hunk 1]
```diff
@@ -0,0 +1,32 @@
+Feature:Social Media Links
+
+  As a site builder
```
Comment (barbun) [inline build/tests/behat/features/social-links.feature]: This ideally should not be tied to a Site Builder role. Can we re-phrase this to "As a user with permission to administer blocks"?

Comment (NigelCunningham) [inline build/tests/behat/features/social-links.feature]: Fixed.

File: build/tests/behat/features/social-links.feature [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  So that minimal configuration for social media links is required out of the box
+
+  @api @javascript @drupal
+  Scenario: The social media links block should be available OOTB
+    Given I am logged in as a user with the "administer blocks" permission and don't need a password change
+    When I go to "/admin/structure/block"
+    Then the "edit-blocks-govcms-social-links-services-region" select list should be set to "sidebar_second"
+    When I follow "edit-blocks-govcms-social-links-services-configure"
+    Then the "Block title" field should contain ""
+    And the "Facebook URL" field should contain "https://www.facebook.com/"
```
Comment (barbun) [inline build/tests/behat/features/social-links.feature]: Is there much value testing the default URLs?

File: build/tests/behat/features/social-links.feature [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    And the "Twitter URL" field should contain "https://twitter.com/"
+    And the "Youtube URL" field should contain "https://www.youtube.com/user/"
+    And the "Vimeo URL" field should contain "http://www.vimeo.com"
+    And the "Flickr URL" field should contain "https://www.flickr.com/"
+    And the "Instagram URL" field should contain "https://www.instagram.com/"
+    And the "Linkedin URL" field should contain "https://www.linkedin.com/"
+    And the "RSS Feed URL" field should contain "rss.xml"
+    And the "Email URL" field should contain "contact"
+    When for "Linkedin URL" I enter "http://example.com/nowhereinparticular"
```
Comment (barbun) [inline build/tests/behat/features/social-links.feature]: Can we rather test all of the fields that are supposed to be available?
Something along the lines of:

And I fill in the following:
| facebook | http://example.com/nowhereinparticular |
| twitter | http://example.com/nowhereinparticular |
| instagram | http://example.com/nowhereinparticular |

File: build/tests/behat/features/social-links.feature [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    And I press "Save block"
+    Then I should see "The block configuration has been saved."
+    When I am on the homepage
+    Then I should see the heading "Connect with us" in the "sidebar_second" region
+    And the response should contain "http://example.com/nowhereinparticular"
```
Comment (barbun) [inline build/tests/behat/features/social-links.feature]: Can we check for specific markup that is set by the module?
E.g. <a href="URL">SERVICE</a>

That way if an image goes missing the test will catch that.

File: build/tests/behat/features/social-links.feature [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  I want to have a configurable block provided
+  So that minimal configuration for social media links is required out of the box
+
+  @api @javascript @drupal
+  Scenario: The social media links block should be available OOTB
+    Given I am logged in as a user with the "administer blocks" permission and don't need a password change
+    When I go to "/admin/structure/block"
+    Then the "edit-blocks-govcms-social-links-services-region" select list should be set to "sidebar_second"
+    When I follow "edit-blocks-govcms-social-links-services-configure"
+    Then the "Block title" field should contain ""
```
Comment (barbun) [inline build/tests/behat/features/social-links.feature]: Should it? It's just not set, we don't need to test for that.

File: build/tests/behat/features/social-links.feature [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    When for "Linkedin URL" I enter "http://example.com/nowhereinparticular"
+    And I press "Save block"
+    Then I should see "The block configuration has been saved."
+    When I am on the homepage
+    Then I should see the heading "Connect with us" in the "sidebar_second" region
+    And the response should contain "http://example.com/nowhereinparticular"
+    And I go to "/admin/structure/block/manage/govcms_social_links/services/configure"
+    When for "Linkedin URL" I enter "https://www.linkedin.com/"
+    And I press "Save block"
+    Then I should see "The block configuration has been saved."
```
Comment (barbun) [inline build/tests/behat/features/social-links.feature]: Why do you need to set it back to what it was? I assume the test would cancel any changes you do. Not 100% sure though :)

File: build/tests/behat/behat.yml [hunk 7]
```diff
@@ -25,6 +25,14 @@ default:
...
       drush:
         root: ../../docroot
+      region_map:
+        page_top: 'div.region-page-top'
+        header: '#header'
+        navigation_bar: 'div.region-navigation'
+        content: '#content'
+        sidebar_second: 'section.region-sidebar-second'
+        footer: 'footer.region-footer'
+        page_bottom: 'div.region-page-bottom'
```
Comment (barbun) [inline build/tests/behat/behat.yml]: Why was all of this added? The test only uses second sidebar.

File: build/tests/behat/features/social-links.feature (hunk 6 above)
Comment (NigelCunningham) [inline build/tests/behat/features/social-links.feature]: When I've been developing tests, I've regularly seen things not getting reset.

File: build/tests/behat/behat.yml (hunk 7 above)
Comment (NigelCunningham) [inline build/tests/behat/behat.yml]: Yes, it does. But I wanted to save time for someone having to re-investigate how regions work later on if/when we want to use them in another scenario.

**Comment by barbun** [comment]
test this

File: build/tests/behat/features/social-links.feature [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+  As a user able to administer blocks
+  I want to have a configurable block provided
+  So that minimal configuration for social media links is required out of the box
+
+  @api @javascript @drupal
+  Scenario: The social media links block should be available OOTB
+    Given I am logged in as a user with the "administer blocks" permission and don't need a password change
+    When I go to "/admin/structure/block"
+    Then the "edit-blocks-govcms-social-links-services-region" select list should be set to "sidebar_second"
```
Comment (barbun) [inline build/tests/behat/features/social-links.feature]: Can we just jump straight onto block configuration page like you do later?
And I go to "/admin/structure/block/manage/govcms_social_links/services/configure"

Then you would verify that it's placed in the right region etc.

File: build/tests/behat/features/social-links.feature [hunk 9]
```diff
@@ -0,0 +1,39 @@
+Feature:Social Media Links
```
Comment (barbun) [inline build/tests/behat/features/social-links.feature]: Can we add a space after the colon for consistency?

**Comment by barbun** [comment]
test this

**Comment by barbun** [comment]
ok to test

**Comment by barbun** [comment]
test this

File: build/tests/behat/behat.yml (hunk 7 above)
Comment (barbun) [inline build/tests/behat/behat.yml]: Please remove the code if it is not relevant to the task.

File: build/tests/behat/features/social-links.feature [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    | instagram | http://example.com/instagram |
+    | linkedin | http://example.com/linkedin |
+    | rss | http://example.com/rss |
+    | email | http://example.com/email |
+    And I press "Save block"
+    Then I should see "The block configuration has been saved."
+    When I am on the homepage
+    Then I should see the heading "Connect with us" in the "sidebar_second" region
+    And the response should contain "<a href=\"http://example.com/facebook\"><img typeof=\"foaf:Image\" src=\"http://govcms.local/profiles/govcms/modules/custom/govcms_social_links/images/facebook.png\" alt=\"Facebook\" title=\"Facebook\"></a>"
+    And the response should contain "<a href=\"http://example.com/twitter\"><img typeof=\"foaf:Image\" src=\"http://govcms.local/profiles/govcms/modules/custom/govcms_social_links/images/twitter.png\" alt=\"Twitter\" title=\"Twitter\"></a>"
```
Comment (barbun) [inline build/tests/behat/features/social-links.feature]: The tests fail because govcms.local is hardcoded.

**Comment by barbun** [comment]
Merged #11

**Pull request merged**